urld58fd5e7a9_longrm_weibo_robot_tgz-pJ8-weibo4j_org_json_TestJ8/src/weibo4j/Weibo.java

```diff
@@ -0,0 +1,3668 @@
+     * @param id The id of the saved search to be retrieved.
+     * @return the data for a saved search
+     * @throws WeiboException when Weibo service or network is unavailable
+     * @since Weibo4J 1.2.0
+     */
+    public SavedSearch showSavedSearch(int id) throws WeiboException {
+        return new SavedSearch(get(getBaseURL() + "saved_searches/show/" + id
+                + ".json", true));
+    }
+
+    /**
+     * Retrieve the data for a saved search owned by the authenticating user specified by the given id.
+     * @return the data for a created saved search
+     * @throws WeiboException when Weibo service or network is unavailable
+     * @since Weibo4J 1.2.0
+     */
+    public SavedSearch createSavedSearch(String query) throws WeiboException {
+        return new SavedSearch(http.post(getBaseURL() + "saved_searches/create.json"
+                , new PostParameter[]{new PostParameter("query", query)}, true));
+    }
+
+    /**
+     * Destroys a saved search for the authenticated user. The search specified by id must be owned by the authenticating user.
+     * @param id The id of the saved search to be deleted.
+     * @return the data for a destroyed saved search
+     * @throws WeiboException when Weibo service or network is unavailable
+     * @since Weibo4J 1.2.0
+     */
+    public SavedSearch destroySavedSearch(int id) throws WeiboException {
+        return new SavedSearch(http.post(getBaseURL() + "saved_searches/destroy/" + id
+                + ".json", true));
+    }
+
+
+	/**
+	 * Obtain the ListObject feed list
+	 * @param uid		User ID or screen_name
+	 * @param listId	The ID or slug of ListObject
+	 * @param auth		if true, the request will be sent with BASIC authentication header
+	 * @return
+	 * @throws WeiboException
+	 */
+	public List<Status> getListStatuses(String uid, String listId, boolean auth) throws WeiboException {
+		StringBuilder sb = new StringBuilder();
+		sb.append(getBaseURL()).append(uid).append("/lists/").append(listId).append("/statuses.xml").append("?source=").append(CONSUMER_KEY);
+		String httpMethod = "GET";
+		String url = sb.toString();
+		//
+		return Status.constructStatuses(http.httpRequest(url, null, auth, httpMethod), this);
+	}
+
+
+	/**
+	 * Obtain ListObject member list
+	 * @param uid		User ID or screen_name
+	 * @param listId	The ID or slug of ListObject
+	 * @param auth		if true, the request will be sent with BASIC authentication header
+	 * @return
+	 * @throws WeiboException
+	 */
+	public UserWapper getListMembers(String uid, String listId, boolean auth) throws WeiboException {
+		StringBuilder sb = new StringBuilder();
+		sb.append(getBaseURL()).append(uid).append("/").append(listId).append("/members.xml").append("?source=").append(CONSUMER_KEY);
+		String httpMethod = "GET";
+		String url = sb.toString();
+		//
+		return User.constructWapperUsers(http.httpRequest(url, null, auth, httpMethod), this);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 2138:2143, 2155:2160.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.